applications/app/controllers/AtomPageController.scala

```diff
@@ -0,0 +1,64 @@
+package controllers
+
+import com.gu.contentapi.client.model.v1.ItemResponse
+import common._
+import model.Cached.{RevalidatableResult, WithoutRevalidationResult}
+import model._
+import play.api.mvc._
+import play.twirl.api.Html
+import scala.concurrent.Future
+import contentapi.ContentApiClient
+import model.content._
+import views.html.fragments.atoms.{ media => MediaAtomBody, storyquestions => StoryQuestionsAtomBody }
+
+class AtomPageController(contentApiClient: ContentApiClient)(implicit context: ApplicationContext) extends Controller with Logging with ExecutionContexts {
+
+  val notYetImplemented = (NotFound, Nil)
+
+  def render(atomType: String, id: String, js: Boolean) = Action.async { implicit request =>
+
+    lookup(s"atom/$atomType/$id") map {
+      case Left(model: MediaAtom) =>
+        (MediaAtomPage(model, withJavaScript = js), MediaAtomBody(model, displayCaption = false, mediaWrapper = Some(MediaWrapper.EmbedPage)))
+      case Left(model: StoryQuestionsAtom) =>
+        (StoryQuestionsAtomPage(model, withJavaScript = js), StoryQuestionsAtomBody(model, isAmp = false))
+      case Left(model: Quiz) =>             notYetImplemented
+      case Left(model: InteractiveAtom) =>  notYetImplemented
+      case Left(model: RecipeAtom) =>       notYetImplemented
+      case Left(model: ReviewAtom) =>       notYetImplemented
+      case Right(other) =>                  (other, Nil)
+    } map {
+      case (atom: AtomPage, body: Html) => renderAtom(atom, body)
+      case (error: Result, _) => renderOther(error)
+    }
```

Comment:
Yeah, I did that to prevent an extra `match` but it is a futile micro-optimisation and makes contracts less clear 👍